src/visual.ts

```diff
@@ -525,114 +563,128 @@ export class DualKpi implements IVisual {
                 break;
         }
 
-        this.titleSize = DualKpi.DefaultTitleSizes[this.sizeCssClass];
-
-        this.updateViewport({
-            width: availableWidth,
-            height: availableHeight
-        });
+        return { chartSpaceBetween, chartTitleSpace, iconOffset };
+    }
 
-        let chartHeight = (availableHeight - (chartSpaceBetween + chartTitleSpace)) / 2;
-        let topChartAxisConfig = { min: data.settings.dualKpiAxis.topChartAxisMin, max: data.settings.dualKpiAxis.topChartAxisMax };
-        let bottomChartAxisConfig = { min: data.settings.dualKpiAxis.bottomChartAxisMin, max: data.settings.dualKpiAxis.bottomChartAxisMax };
+    private dropBottomChart({
+        data,
+        chartHeight,
+        chartSpaceBetween,
+        chartWidth,
+    }: {
+        data: IDualKpiData
+        chartHeight: number;
+        chartSpaceBetween: number;
+        chartWidth: number;
+    }) {
+        if (data.settings.dualKpiProperties.bottomChartShow.value) {
+            const bottomChartAxisConfig = { min: data.settings.dualKpiAxis.bottomChartAxisMin.value, max: data.settings.dualKpiAxis.bottomChartAxisMax.value };
+            const bottomChartPercentChangeStartPoint = DualKpi.getPercentChangeStartPoint(data.bottomValues, data.bottomPercentCalcDate);
 
-        let topChartPercentChangeStartPoint = DualKpi.getPercentChangeStartPoint(data.topValues, data.topPercentCalcDate);
-        let bottomChartPercentChangeStartPoint = DualKpi.getPercentChangeStartPoint(data.bottomValues, data.bottomPercentCalcDate);
+            this.chartGroupBottom.group.classed(DualKpi.INVISIBLE, false);
+            this.chartGroupBottom.hoverDataContainer.container.classed(DualKpi.INVISIBLE, true);
 
-        const wasDataSetRendered: boolean = data.topValues.length > 0 || data.bottomValues.length > 0;
+            if (data.bottomValues.length > 0) {
+                this.drawChart({
```

Comment:
move such a big argument object to a variable outside, to make the function itself clearer and easier to read.